searx/engines/torznab.py

```diff
@@ -66,78 +74,131 @@ def response(resp):
+
+    def _map_result_url(guid: Optional[str], comments: Optional[str]) -> Optional[str]:
+        if guid is not None and guid.startswith("http"):
+            return guid
+        if comments is not None and comments.startswith("http"):
+            return comments
+        return None
+
+    def _map_leechers(
+        leechers: Optional[str], seeders: Optional[str], peers: Optional[str]
+    ) -> Optional[str]:
+        if leechers is not None:
+            return leechers
+        if seeders is not None and peers is not None:
+            return str(int(peers) - int(seeders))
+        return None
+
+    def _map_published_date(pubDate: Optional[str]) -> Optional[datetime]:
+        if pubDate is not None:
+            try:
+                return datetime.strptime(pubDate, "%a, %d %b %Y %H:%M:%S %z")
+            except (ValueError, TypeError) as e:
+                logger.debug("ignore exception (publishedDate): %s", e)
+        return None
+
+    def _map_torrent_file(
+        link: Optional[str], enclosure_url: Optional[str]
+    ) -> Optional[str]:
+        if link is not None and link.startswith("http"):
+            return link
+        if enclosure_url is not None and enclosure_url.startswith("http"):
+            return enclosure_url
+        return None
+
+    def _map_magnet_link(
+        magneturl: Optional[str],
+        guid: Optional[str],
+        enclosure_url: Optional[str],
+        link: Optional[str],
+    ) -> Optional[str]:
+        if magneturl is not None and magneturl.startswith("magnet"):
+            return magneturl
+        if guid is not None and guid.startswith("magnet"):
+            return guid
+        if enclosure_url is not None and enclosure_url.startswith("magnet"):
+            return enclosure_url
+        if link is not None and link.startswith("magnet"):
+            return link
+        return None
+
+    # extract attributes from XML
+    # see https://torznab.github.io/spec-1.3-draft/torznab/Specification-v1.3.html#predefined-attributes
+    enclosure: Optional[etree.Element] = item.find("enclosure")
+    enclosure_size: Optional[str]
+    enclosure_url: Optional[str]
+    if enclosure is not None:
+        enclosure_url = enclosure.get("url")
+        enclosure_size = enclosure.get("length")
+
+    link: Optional[str] = get_attribute(item, "link")
+    guid: Optional[str] = get_attribute(item, "guid")
+    comments: Optional[str] = get_attribute(item, "comments")
+    title: Optional[str] = get_attribute(item, "title")
+    pubDate: Optional[str] = get_attribute(item, "pubDate")
+    size: Optional[str] = int(get_attribute(item, "size"))
+    seeders: Optional[str] = get_torznab_attribute(item, "seeders")
+    leechers: Optional[str] = get_torznab_attribute(item, "leechers")
+    peers: Optional[str] = get_torznab_attribute(item, "peers")
+    files: Optional[str] = get_attribute(item, "files")
+    magneturl: Optional[str] = get_torznab_attribute(item, "magneturl")
```

Comment:
IMO this annotation are not needed.

Reply:
Done in [9253377](https://github.com/searxng/searxng/pull/2514/commits/9253377b2ac519c1dd02d18b91f4bff4ccc6d54c)